### PR TITLE
Wok 410 proxy deepzoom

### PR DESCRIPTION
**Pull request by VictorHarbo**

### CODE REVIEW

File: conf/ds-image-environment.yaml.sample [hunk 1]
```diff
@@ -13,7 +13,9 @@
...
 
   # Production machine setup
-  backend:
-    db: 'http://realmachine:8090/foobar/'
-    user: 'real_dbuser'
-    password: 'real_dbpassword'
+  imageservers:
+    # Full request URL sample: http://example.com:1234/iipsrv/iipsrv.fcgi?FIF=/my_images/hello.tif&CVT=jpg
+    iip: 'http://example.com:1234/iipsrv/iipsrv.fcgi'
+
```
Comment (tokee) [inline conf/ds-image-environment.yaml.sample:19]: Please add sample (i.e. non-working URL) for Deepzoom

File: src/main/java/dk/kb/image/IIPFacade.java [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+
+    public javax.ws.rs.core.StreamingOutput getDeepzoomTile(
+            URI requestURI,
+            String imageid, Integer layer, String tiles, String format, Float CNT, List<Integer> SHD,
+            Float GAM, String CMP, String CTW, Boolean INV, String COL) throws ServiceException {
+        validateDeepzoomTileRequest(imageid, layer, tiles, format, CNT, SHD, GAM, CMP, CTW, INV, COL);
+
+        // Defaults
+        if (format == null || "jpg".equals(format)) {
+            format = "jpeg";
```
Comment (tokee) [inline src/main/java/dk/kb/image/IIPFacade.java:166]: We cannot assume that the backend always wants `jpeg`over `jpg` for Deepzoom requests, so we should only set the format to `jpeg` if it is null.

File: src/main/java/dk/kb/image/IIPFacade.java [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        }
+
+        // TODO: Use the UriTemplate system like IIIFFacade
+        // http://example.com//fcgi-bin/iipsrv.fcgi?DeepZoom=/your/image/path.tif.dzi
+        UriBuilder builder = UriBuilder.
+                fromUri(ServiceConfig.getConfig().getString(KEY_DEEPZOOM_SERVER)).
+                path(imageid + "_files").path("" + layer).path(tiles + "." + format); // Mandatory
```
Comment (tokee) [inline src/main/java/dk/kb/image/IIPFacade.java:173]: Sudden realization: Deepzoom can be accessed both as path and as query params. We need to handle this in the configuration & code. We need to discuss how to do so.

**Review by tokee — CHANGES_REQUESTED** [review]
We have a problem with the Deepzoom endpoint as it can be either path- or query param driven

File: conf/ds-image-environment.yaml.sample (hunk 1 above)
Comment (VictorHarbo) [reply conf/ds-image-environment.yaml.sample:20]: Samples has been added

File: src/main/java/dk/kb/image/IIPFacade.java (hunk 2 above)
Comment (VictorHarbo) [reply src/main/java/dk/kb/image/IIPFacade.java:180]: It only changes null now

**Pull request merged**